compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -82,71 +82,423 @@ static void lookupScheme2(TR::Node *node, bool unbalanced, bool fromTableEval, T
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_LT << TR::RealRegister::pos_RB));
+         generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::cror, node, condReg, condReg, condReg2,
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_GT << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RB));
+         break;
+
+      case CompareCondition::gt:
+         // x_h > y_h || (x_h == y_h && x_l > y_l)
+         generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::crand, node, condReg2, condReg, condReg2,
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_GT << TR::RealRegister::pos_RB));
+         generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::cror, node, condReg, condReg, condReg2,
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_GT << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RB));
+         break;
+
+      case CompareCondition::le:
+         // x_h < y_h || (x_h == y_h && x_l <= y_l)
+         generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::crandc, node, condReg2, condReg, condReg2,
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_GT << TR::RealRegister::pos_RB));
+         generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::cror, node, condReg, condReg, condReg2,
+            (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RT) | (TR::RealRegister::CRCC_LT << TR::RealRegister::pos_RA) | (TR::RealRegister::CRCC_EQ << TR::RealRegister::pos_RB));
+         break;
+      }
+
+   cg->stopUsingRegister(condReg2);
+   return CompareCondition::eq;
+   }
+
+CompareCondition evaluateIntCompareToConditionRegister(
+   TR::Register *condReg,
+   TR::Node *node,
+   TR::Node *firstChild,
+   TR::Node *secondChild,
+   const CompareInfo& compareInfo,
+   TR::CodeGenerator *cg
+)
+   {
+   TR::InstOpCode::Mnemonic cmpOp;
+   TR::InstOpCode::Mnemonic cmpiOp;
+
+   bool is64Bit;
+
+   switch (compareInfo.type)
+      {
+      case TR::Int8:
+      case TR::Int16:
+      case TR::Int32:
+         is64Bit = false;
+         break;
+      case TR::Int64:
+         is64Bit = true;
+         break;
+      case TR::Address:
+         is64Bit = cg->comp()->target().is64Bit();
+         break;
+      default:
+         TR_ASSERT_FATAL_WITH_NODE(node, false, "Cannot call evaluateIntCompareToConditionRegister with data type %s", TR::DataType::getName(compareInfo.type));
+      }
+
+   if (is64Bit && !cg->comp()->target().is64Bit())
+      return evaluateDualIntCompareToConditionRegister(condReg, node, firstChild, secondChild, compareInfo, cg);
```

Comment:
lost the optimization where seocndChild is a constant which can be handled more efficiently.

Reply:
I don't see any such optimization in the previous code for this (`iflcmp*` has some handling for this, but `lcmp*` never did), but you make a good point regardless. I've gone ahead and added exploitation of `cmplwi`/`cmpwi` for constants where applicable.